prac_09/sort_files.py

```diff
@@ -0,0 +1,41 @@
+"""
+CP1404/CP5632 Practical
+Program to sort files into directories based on file type
+By Christopher Caferra
+"""
+
+import os
+import shutil
+
+
+def main():
+    """Sort files into directories based on file extension"""
+    extensions = []
+    extension = ''
+    os.chdir('FilesToSort')
+    for filename in os.listdir('.'):
+        file_type = filename[filename.find('.') + 1:]
+        extensions.append(file_type)
+        for extension in extensions:
+            if not os.path.isdir(extension):
+                os.mkdir(extension)
+        try:
+            shutil.move(filename, extension + '/' + filename)
+        except shutil.Error:
+            pass
```

Comment:
You could possibly change all the code above (to the next comment) to
```python
if not os.path.isdir(filename):
    try:
        os.mkdir(file_type)
    except FileExistsError:
        pass
shutil.move(filename, file_type + '/' + filename)
```
which would only try to create the last extension folder (and not if the folder already exists) as you only need to process the last extension (all previous have already been added). Extensions as a result